### PR TITLE
Updates 1.8/usage/registry

### PR DESCRIPTION
**Pull request by GoelDeepak**
## What type of changes does your doc introduce?
- [x] Bug fix
- [ ] New feature 

## What is the urgency level of this change?
- [ ] Blocker
- [ ] High
- [x] Medium

## I have completed these items:
- [x] I have tested all commands and code snippets.
- [x] I have built locally and tested for broken links and formatting.
- [x] I have made this change for all affected versions (e.g. 1.7, 1.8, and 1.9).
- [x] My doc follows the style guide: https://github.com/dcos/dcos-docs#contributing.

Ping `@emanic`, `@sascala`, or `@joel-hamill` for reviewing pull request changes.

### CODE REVIEW

**Comment by GoelDeepak** [comment]
@joel-hamill  ready for review

File: 1.8/usage/registry.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+post_title: Deploying a private Docker registry using VIPs
+nav_title: Private Docker Registry
+menu_order: 40
+---
+
+You can use a number of (hosted) [options](https://mesosphere.com/blog/2015/10/14/docker-registries-the-good-the-bad-the-ugly/) to store and manage your Docker images, but you can also set up a private Docker registry yourself and run it within your DC/OS cluster. This private registry is based on the [Docker registry image](https://docs.docker.com/registry/) and stores images in a system using one of the available storage drivers.
+
+The advantages of a private Docker registry are:
+
+- They can be securely deployed in your local environment (DC/OS cluster).
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: Suggest: `They can be securely deployed in your DC/OS cluster.`

File: 1.8/usage/registry.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+- Reduced latency for pushing and pulling images, leading to faster deployments.
+
+This article describes how to set up a private Docker registry in a DC/OS cluster and make it accessible through a [VIP](docs/1.8/usage/service-discovery/load-balancing-vips/virtual-ip-addresses/).
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: Link should be: `/docs/1.8...`

File: 1.8/usage/registry.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+1. __Insecure__ This option should be avoided when possible. It may even conflict with Docker-in-Docker containers such as Jenkins.
+
+For more information on setting up a Docker Trusted registry see [deploying a registry server](https://docs.docker.com/registry/deploying/) via the Docker docs.
+
+## Naming and discovery options
+
+After the registry is deployed clients will need to access it. When you create the certificate, you should consider the naming and discovery approach used to access the service.  You can use any of the [service discovery](/docs/1.8/usage/service-discovery/) options available in DC/OS:
+
+- __Mesos-DNS__ with host port mapping is an easy option, but requires that ports are managed as resources in the cluster.
+- __Minuteman__ provides a stable endpoint and you can use either the VIP or a DNS entry that references the VIP. A DNS entry can add a bit more complexity but improves consumption of the service.
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: Change `Minuteman` to `Virtual IP`. VIP is the official feature name, not minuteman.

File: 1.8/usage/registry.md (hunk 2 above)
Comment (joel-hamill) [inline 1.8/usage/registry.md]: `This tutorial...`

File: 1.8/usage/registry.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+## Deploy a Docker registry using a self-signed certificate
+
+At a high level, here are the three steps required to set up a private Docker registry:
+
+1. Create a self-signed certificate with the correct common name and subject alternative names.
+1. Configure the Docker Registry service with the public and private key for the certificate.
+1. Configure Docker daemons to trust the self-signed certificates.
+
+Open an SSH connection to a master node in the cluster with credentials forwarding enabled (option `-A`), so that an SSH connection can be made to other nodes in the cluster. Note: this tutorial uses [jq](https://stedolan.github.io/jq/), but you can also carry out the steps using, for example, `grep`.
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: Suggest `Note: this tutorial uses [jq](https://stedolan.github.io/jq/), but you can also carry out the steps using, for example, `grep`.` to `This tutorial uses [jq](https://stedolan.github.io/jq/), but you can complete the steps using other methods, for example, `grep`.`

File: 1.8/usage/registry.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+
+```bash
+$ for i in $(curl -sS master.mesos:5050/slaves | jq '.slaves[] | .hostname' | tr -d '"'); do ssh "$i" -oStrictHostKeyChecking=no "sudo sysctl -w net.netfilter.nf_conntrack_tcp_be_liberal=1"; done
+```
+
+### Create a self signed cert
+
+The first step is to create a self-signed certificate to secure a Docker registry. It’s important that the common name used in the certificate matches the domain name that is used to access the registry.
+
+You can create a self-signed certificate to secure a registry by using OpenSSL. This command creates a certificate that uses a minuteman VIP to access the registry.
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: Are there other types of DC/OS VIPs, besides minutemen? If not, change to `...uses a VIP to access...`. If so, change to `...uses a Minuteman VIP to access...`

File: 1.8/usage/registry.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+You can use the following commands to copy the certificate and keys to all agents in the cluster:
+
+```bash
+$ MESOS_AGENTS=$(curl -sS master.mesos:5050/slaves | jq '.slaves[] | .hostname' | tr -d '"');
+$ for i in $MESOS_AGENTS; do ssh "$i" -oStrictHostKeyChecking=no "sudo mkdir --parent /etc/privateregistry/certs/"; done
+$ for i in $MESOS_AGENTS; do scp -o StrictHostKeyChecking=no ./domain.* "$i":~/; done
+$ for i in $MESOS_AGENTS; do ssh "$i" -oStrictHostKeyChecking=no "sudo mv ./domain.* /etc/privateregistry/certs/"; done
+```
+
+You must configure the Docker daemon on all machines that require registry access to trust the self-signed certificate. You must  create a folder that matches the FQDN in the `certs.d` folder and copy the certificate to the folder using the name `ca.crt`. You can use the following script to create the folders and copy the files into the appropriate locations:
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: Use this script to:

* Configure the Docker daemon on all machines that require registry access to trust the self-signed certificate and create a folder that matches the FQDN in the `certs.d` folder.
* Copy the certificate to the folder using the name `ca.crt`.`

File: 1.8/usage/registry.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+
+$ dcos marathon app add registry.json
+```
+
+## Test the deployment
+
+Now that the registry is deployed you can push an image and deploy a new Marathon application by using it as follows:
+
+1. Connect to one of the nodes that has the Docker daemon configured to trust the self-signed certificate.
+1. Pull the `flakio httpenv` image from docker hub using the following command: `sudo docker pull flakio/httpenv:1`
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: ...the following command: 

```bash
sudo docker pull flakio/httpenv:1
```

File: 1.8/usage/registry.md [hunk 8]
```diff
@@ -0,0 +1,167 @@
...
+1. Add a new tag to the image allowing us to push it to our private registry: `sudo docker tag flakio/httpenv:1 registry.marathon.l4lb.thisdcos.directory:5000/httpenv:1`
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: ...our private registry: 

```bash
sudo docker tag flakio/httpenv:1 registry.marathon.l4lb.thisdcos.directory:5000/httpenv:1
```

File: 1.8/usage/registry.md [hunk 9]
```diff
@@ -0,0 +1,167 @@
...
+1. Push the image to the registry we just deployed: `sudo docker push registry.marathon.l4lb.thisdcos.directory:5000/httpenv:1`
```
Comment (joel-hamill) [inline 1.8/usage/registry.md]: ...we just deployed: 

```bash
sudo docker push registry.marathon.l4lb.thisdcos.directory:5000/httpenv:1
```

**Comment by mhausenblas** [comment]
Why are we duplicating work? It exists already in https://github.com/dcos/examples/tree/master/1.8/registry unless I'm missing something here? @GoelDeepak can you check pls?

**Comment by GoelDeepak** [comment]
@mhausenblas I agree with you. However, not sure how we expose document on github vs on our website. In the past, I have seen duplicate documents both on github and our website. @joel-hamill what are your thoughts?

**Comment by mhausenblas** [comment]
@GoelDeepak Docs team and OSS team agreed to move the tutorials to dcos/examples. Suggest you talk with @williamsaaron …

**Comment by joel-hamill** [comment]
@mhausenblas @GoelDeepak think we need to figure out way to link to this directly from the left-hand nav too (https://cl.ly/332f3C3V3H0n). as i see it we have 2 ways to do this:

- add functionality to manually add links to the left-hand navigation (CC: @ashenden) 
- add empty usage/tutorials/registry.md topic with a redirect to the https://github.com/dcos/examples/tree/master/1.8/registry page

the first option is probably faster to implement.

**Comment by GoelDeepak** [comment]
@joel-hamill Should I drop this PR?

**Comment by joel-hamill** [comment]
@GoelDeepak let me ping some folks and figure out which option to take.

**Comment by joel-hamill** [comment]
Added redirect for 1.8/usage/registry: https://github.com/dcos/dcos-website/pull/988

**Comment by GoelDeepak** [comment]
Cool. I am dropping this PR

**Pull request closed without merging**